Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -278,49 +280,82 @@ void WebCookieJar::deleteCookie(const WebCore::Document& document, const URL& ur
 
 void WebCookieJar::getCookiesAsync(WebCore::Document& document, const URL& url, const WebCore::CookieStoreGetOptions& options, CompletionHandler<void(std::optional<Vector<WebCore::Cookie>>&&)>&& completionHandler) const
 {
-    RefPtr webFrame = document.frame() ? WebFrame::fromCoreFrame(*document.frame()) : nullptr;
-    if (!webFrame || !webFrame->page()) {
+    auto frame = document.frame();
+    if (!frame) {
         completionHandler({ });
         return;
     }
 
+    std::optional<WebCore::FrameIdentifier> frameID = std::nullopt;
+    std::optional<WebCore::PageIdentifier> pageID = std::nullopt;
+
     ApplyTrackingPrevention applyTrackingPreventionInNetworkProcess = ApplyTrackingPrevention::No;
 #if ENABLE(TRACKING_PREVENTION)
-    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), url, applyTrackingPreventionInNetworkProcess)) {
-        completionHandler({ });
-        return;
+    if (frame->loader().client().isRemoteWorkerFrameLoaderClient()) {
```

Comment:
I don't think we need to special case remote workers in this function. We can just rely on the fact that there is a WebFrame or not.